cli/Flags.cpp

```diff
@@ -87,4 +87,25 @@ DEFINE_bool(preload_buffer_pool, false,
             "accepting queries (should also set --buffer_pool_slots to be "
             "large enough to accomodate the entire database).");
 
+static bool ValidatePort(const char *flagname, std::int32_t value) {
+  int const min = 0, max = 65536;
+  if (value > min && value < max) {
+    return true;
+  }
+  std::cout << "Invalid value for --" << flagname << ": " << value
+            << "\nUse ports between " << min << " and "
+            << max << std::endl;
+  return false;
+}
+
+DEFINE_int32(cli_network_port, 3000,
+             "Listens for TCP connections on this port when network mode is enabled. "
+             "This is only used if the cli is set to use the network mode (--mode=network).");
+DEFINE_validator(cli_network_port, &ValidatePort);
+
+DEFINE_string(cli_network_ip, "localhost",
```

Comment:
Also the CliClientMain uses the flag so I think they are appropriate to be kept there.